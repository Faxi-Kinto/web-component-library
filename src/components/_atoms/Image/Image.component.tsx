import React from 'react';
import * as Styled from './Image.styles';

/**
 * @name Image
 *
 * [Insert Component Description]
 *
 * @returns {JSX}
 */

const defaultWidth = 'auto';
const defaultHeight = 'auto';

type ImageObjectFit =
  | 'contain'
  | 'cover'
  | 'fill'
  | 'inherit'
  | 'initial'
  | 'none'
  | 'scale-down'
  | 'unset';

export type ImageProps = {
  src: string;
  alt: string;
  width?: string;
  height?: string;
  objectFit?: ImageObjectFit;
  className?: string;
  onClick?: Function;
};

const Image: React.FC<ImageProps> = (props: ImageProps): JSX.Element => {
  const { src, alt, width, height, objectFit, className, onClick } = props;

  return (
    <Styled.Image
      src={src}
      alt={alt}
      width={width || defaultWidth}
      height={height || defaultHeight}
      className={className}
      style={{ objectFit }}
      onClick={onClick && onClick()}
    />
  );
};

export default Image;
