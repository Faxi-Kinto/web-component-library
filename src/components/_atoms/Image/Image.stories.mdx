import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Image from './';

<Meta title="Core|Atoms/Image" />

# Image Component

## Implementation

#### Default

<Preview>
  <Story name="default">
    <Image
      src="https://live.staticflickr.com/2084/2222523486_5e1894e314_b.jpg"
      alt="Image"
    />
  </Story>
</Preview>

# Props

<Props of={Image} />
