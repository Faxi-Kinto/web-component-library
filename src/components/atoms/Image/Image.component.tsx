import React from 'react';
import * as Styled from './Image.styles';
import { pxToRem } from '@faxi/web-css-utilities';

/**
 * @name Image
 *
 * [Insert Component Description]
 *
 * @returns {JSX}
 */

const defaultWidth = pxToRem('95px');
const defaultHeight = pxToRem('95px');

type ImageObjectFit =
  | 'contain'
  | 'cover'
  | 'fill'
  | 'inherit'
  | 'initial'
  | 'none'
  | 'scale-down'
  | 'unset';

export type ImageProps = {
  src: string;
  alt: string;
  width?: string;
  height?: string;
  objectFit?: ImageObjectFit;
  className?: string;
};

const Image: React.FC<ImageProps> = (props: ImageProps): JSX.Element => {
  const { src, alt, width, height, objectFit, className } = props;

  return (
    <Styled.Image
      src={src}
      alt={alt}
      width={width || defaultWidth}
      height={height || defaultHeight}
      className={className}
      style={{ objectFit }}
    />
  );
};

export default Image;
