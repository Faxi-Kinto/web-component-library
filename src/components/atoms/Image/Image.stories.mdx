import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Image from './';

<Meta title="Core|Atoms/Image" />

# Image Component

## Implementation

#### Default

<Preview>
  <Story name="default">
    <Image src="" alt="Image" />
  </Story>
</Preview>

# Props

<Props of={Image} />
