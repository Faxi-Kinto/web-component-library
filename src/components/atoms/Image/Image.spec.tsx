import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image.component';

test('Image renders correctly', () => {
  const component = renderer.create(
    <Image src="/assets/images/albert.jpg" alt="avatar" />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
