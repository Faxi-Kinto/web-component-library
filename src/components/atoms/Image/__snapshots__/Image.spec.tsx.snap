// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
.c0 {
  width: auto;
  height: auto;
}

<img
  alt="avatar"
  className="c0"
  height="auto"
  src="/assets/images/albert.jpg"
  style={
    Object {
      "objectFit": undefined,
    }
  }
  width="auto"
/>
`;
